### ISSUE TITLE
TCPDF ERROR: [Image] Unable to get the size of the image when trying to export a ticket to PDF

### ISSUE THREAD

**Issue by EusebioGM** [yllen/pdf#21]
**BUG DESCRIPTION**

"**PDF PRINT**: TCPDF ERROR: [Image] Unable to get the size of the image: https://localhost/glpi/plugins/pdf/front/export.php"

Hi support team, i have an issue with the lastest pdf plugin (2.1.0) with GLPI 10.0.1 (downloaded from "https://plugins.glpi-project.org/#/plugin/pdf" ) when trying to export a ticket to PDF.

**IMPORTANT:**
It only occurs when "Ticket" has attached  images, or emails on it--> If I try to generate a PDF file from a ticket that has no e-mails/images attached, the plugin works fine.

![image](https://user-images.githubusercontent.com/26650858/185461579-0ee11164-df0d-41ee-aed5-2eb0523b3766.png)

Could you please help me with this issue, please?
I can sent you logs if you tell me the name of lof file, paths, commands to execute (if necessary)
I remember when I was working with GLPI v9.5 (and pdf v2.0) it worked fine. An it was fantastic!

**CLARIFICATION:**
I had the SAME problem since I updated to GLPI v10.0.0 with the lastest "pdf plugin (2.1.0)"
**_In other words, the problem already existed with GLPI v10.0.0 with "pdf plugin (2.1.0)"_**

So, I tried to update to **GLPI** **v10.0.1** to see if the upgrade to the latest version coud fix that problem with the plugin, but the problem persists.
The PDF conversion plugin is a really fantastic plugin when it is working fine. 
I would strongly appreciate it if you could help me solve with this issue.

Thanks in advance!

**Comment by yllen** [comment]
The system change in GLPI since 10.0.0.
Can you test my solution and confirm it's OK for you?

**Comment by EusebioGM** [comment]
Hi Yllen,

Thanks a lot for your post.
I have just applied the fix and I think we have had some progress with your solution.
Now I have detected two different CASES depending on the tickets that have images embedded inside the ticket:

**CASE 1) The pdf convertion works (in same tickets with several images embedded):**
But when it works the images in the pdf file are not displayed (same than "https://github.com/yllen/pdf/issues/19"; so maybe this issue was solved for glpi 95 but still not solved for GLPI10.0.1 with "pdf plugin 2.1.0"). 

**CASE 2) The pdf convertion does not work in other tickets (with several images embedded):**
It keeps the same error format as you can see in the following picture:

![image](https://user-images.githubusercontent.com/26650858/191935829-b6a41f38-46a9-4eb3-b863-d70284992003.png)

I copy the error message: "TCPDF ERROR: [Image] Unable to get the size of the image: https://if-of-glpi/glpi/front/document.send.php?docid=1713&tickets_id=373"

Could you please help me?
Thanks a lot,

**Comment by yllen** [comment]
In GLPI10, documents are not saved in the database as in GLPI95 and i don't find a script to change this.
So, i need more information:
- have you some errors in files/_log/php-error.log and sql-error.log
- case 1: in database, glpi_tickets, post me the result of the content column
- case 2: the document id 1713 still exists in your GLPI? can you open it from document menu of GLPI?

**Comment by EusebioGM** [comment]
Thanks a lot for your answer:
Here you have my screenshots:

* No errors in both files "/_log/php-error.log" and "/_log/sql-error.log"
![image](https://user-images.githubusercontent.com/26650858/192266868-268d409b-4306-4089-8714-75988f10e186.png)
![image](https://user-images.githubusercontent.com/26650858/192266884-140c9e25-3481-4ac2-8039-f147536e72a5.png)

* case 1: in database, glpi_tickets, post me the result of the **content** column
![image](https://user-images.githubusercontent.com/26650858/192265495-f1b8c812-d697-48c7-acbc-0d1469e13ee1.png)
* Case2: the document id 1713 does exist in my GLPI. And I confirm I can you open it from document menu of GLPI.
And I confirm I can open it if I type the full URL in "Firefox or Chrome Explorers"
![image](https://user-images.githubusercontent.com/26650858/192265989-0ee2ea09-60f3-47b4-a457-291cdbb130db.png)
![image](https://user-images.githubusercontent.com/26650858/192265743-a00ff07d-42b4-4dff-a59f-e0a33aea411e.png)

Let me know If If you need more information on my side.
Thanks a lot!!

**Comment by EusebioGM** [comment]
Ok, Thanks a lot!!
Here you have the content column for CASE1 :

&lt;p&gt;SMG envia este correo  (&lt;a href="https://ip-of-glpi/glpi/front/document.send.php?docid=2342&amp;tickets_id=477" target="_blank" rel="noopener"&gt;Depuración de reglas.msg&lt;/a&gt;) tras la reunión con Dgo, Hg, Pedri y yo.&lt;/p&gt;&lt;p&gt;******************************************************&lt;/p&gt;&lt;p&gt;Hola a todos,&lt;/p&gt;&lt;p&gt;Como todos sabéis, desde hace tiempo venimos arrastrando una serie de “prácticas” (por decirlo de alguna manera), a la hora de implementar reglas.&lt;/p&gt;&lt;p&gt;Esto quedó de nuevo a la vista cuando vimos vía Teams el asunto de incluir los nuevos equipos en el grupo “Estaciones1”, etc …&lt;/p&gt;&lt;p&gt;Para poner remedio a este problema, es necesario revisar y depurar todas las políticas que aplican, etc&lt;/p&gt;&lt;p&gt;Si os parece bien, vamos a comenzar a depurar las políticas que afectan a un grupo; aquí he filtrado las reglas donde existe un “valor” en el campo “Services &amp; Applications” (el firewall deja pasar los puertos) , y os lo adjunto en un Excel (&lt;a href="https://ip-of-glpi/glpi/front/document.send.php?docid=2341&amp;tickets_id=477" target="_blank" rel="noopener"&gt;20220404_fase1_reglas.xlsx&lt;/a&gt;).&lt;/p&gt;&lt;p&gt;En el excel veréis estas reglas con varios colores de fondo, en función de si la depuración de las reglas corresponde en principio a equipo1 (fondo azul), a equipo2 (fondo verde), o a equipo3 (fondo salmón) .&lt;/p&gt;&lt;p&gt;La idea es que cada departamento revise los puertos que han de ir asociados a sus reglas correspondientes, y una vez identificados, hagáis una petición por la otra herramienta, para ir poco a poco dejando las reglas como deben estar (con los puertos finales).&lt;/p&gt;&lt;p&gt;Una vez tengamos esto depurado, continuaremos por ejemplo con las reglas que tienen como destino = valor, e iremos así sucesivamente depurando, hasta dejar las reglas en un estado coherente.&lt;/p&gt;&lt;p&gt;Si tenéis alguna duda o comentario, podemos tener una reunión por Teams.&lt;/p&gt;&lt;p&gt;Un saludo,&lt;/p&gt;&lt;p&gt;Serio&lt;/p&gt;&lt;p&gt;******************************************************&lt;/p&gt;

**Comment by yllen** [comment]
can you post the content for ticket ID 373?

**Comment by yllen** [comment]
In your example, you have a link to a picture of a ticket not and url to display the picture.
To display the picture i need and img tag with srv reference

**Comment by yllen** [comment]
Trying with your ticket's content
- in GLPI a link is displayed
![image](https://user-images.githubusercontent.com/12936722/195128026-17485a08-bfde-4cb5-85e0-c0333fc1a9d8.png)

- in PDF a link is displayed to
-
![image](https://user-images.githubusercontent.com/12936722/195128151-87daf4ad-7549-4766-a10e-a860df247a80.png)

**Comment by EusebioGM** [comment]
> can you post the content for ticket ID 373?

Here you hace "Content (ID 373)":

&lt;p&gt;Servidor ePO bloquea los propios proceso de McAfee&lt;/p&gt;&lt;p&gt;Existe un BUG y un script "bat"  para instalar un certificado propio de McAfee que evita que los propios productos de McAfee se bloquéen por el mismo VSE&lt;/p&gt;&lt;p&gt;&lt;a href="https://kc.mcafee.com/corporate/index?page=content&amp;id=KB12345"&gt;https://kc.mcafee.com/corporate/index?page=content&amp;id=KB12345&lt;/a&gt;&lt;/p&gt;&lt;p&gt; &lt;/p&gt;&lt;p&gt;&lt;a href="https://x.x.x.x/glpi/front/document.send.php?docid=1705&amp;tickets_id=373" target="_blank" rel="noopener"&gt;&lt;img src="https://x.x.x.x/glpi/front/document.send.php?docid=1705&amp;tickets_id=373" alt="ca2881ed-d9d62df8-619e65587ea463.90223236" width="649" /&gt;&lt;/a&gt;&lt;/p&gt;

**Comment by EusebioGM** [comment]
Hi Yllen, I do not know if I understand you correctly.
Here you have ALL the contents fields of the Ticket 373 (This is the HEADER of the ticket):

'373', '0', 'Servidor bloquea los propios procesos, '2021-11-08 15:38:50', NULL, NULL, '2022-03-14 18:10:27', '6', '4', '6', '0', '&lt;p&gt;Servidor ePO bloquea los propios proceso de McAfee&lt;/p&gt;&lt;p&gt;Existe un BUG y un script \"bat\"  para instalar un certificado propio de que evita que los propios productos se bloquéen por el mismo &lt;/p&gt;&lt;p&gt;&lt;a href=\"https://kc.web.com/corporate/index?page=content&amp;id=KB12345\"&gt;https://kc.ewb.com/corporate/index?page=content&amp;id=KB12345&lt;/a&gt;&lt;/p&gt;&lt;p&gt; &lt;/p&gt;&lt;p&gt;&lt;a href=\"https://x.x.x.x/glpi/front/document.send.php?docid=1705&amp;tickets_id=373\" target=\"_blank\" rel=\"noopener\"&gt;&lt;img src=\"https://x.x.x.x/glpi/front/document.send.php?docid=1705&amp;tickets_id=373\" alt=\"ca2881ed-d9d62df8-619e65587ea463.90223236\" width=\"649\" /&gt;&lt;/a&gt;&lt;/p&gt;', '3', '3', '3', '0', '1', '1', '0', '0', '0', NULL, NULL, '2022-03-14 18:10:27', '0', '0', '0', '0', '0', NULL, NULL, NULL, '618106', '0', '0', '1', '0', '0', '11', '0', '2021-11-08 15:38:50'

Additionally, here you have a picture about the 5 rows (FollowUps) from the same ticket ID. 
![image](https://user-images.githubusercontent.com/26650858/197573418-dd8f736d-795b-45c1-bf7b-1b320affe7ae.png)

(I can not sent you "here" the total content of the rows because their content. Anyway, if you need the content, instead of the picture, you can tell me how to send them to you as a private message.)

Thanks a lot!

**Comment by EusebioGM** [comment]
I have several pics.
One picture is included in the ticket HEADER (first one with green colour in the background).
The other ones are in the FollowUps ot the ticket (second picture "with a lot of characters" in gray colour)
![image](https://user-images.githubusercontent.com/26650858/198052291-c841e173-be1f-43ed-88f0-e6c32a49e97c.png)

Additionally, you can see more pics in the next FollowUps ot the ticket (as well in gray colour).

![image](https://user-images.githubusercontent.com/26650858/198056708-b214a150-0011-4022-ba35-23c98757e122.png)

The last picture in my previous image (the only one in the form of a table and with sequential dates), is the one that leads us to the error when I try to export the whole ticket (HEADER and content/FollowUps) to pdf with the error you already know:
 "TCPDF ERROR: [Image] Unable to get the size of the image: https://if-of-glpi/glpi/front/document.send.php?docid=1713&tickets_id=373"

**Comment by yllen** [comment]
sent me
- content of follouwup of this ticket
- content of this ticket
- picture of this ticket in GLPI
- picture of your followup
- exactly error massage in GLPI log
glpi@ll-experts.com

**Comment by EusebioGM** [comment]
I have sent the email withe info. you requested to glpi@ll-experts.com
Thanks a lot!!

**Comment by stdglpi** [comment]
Hello all,
I follow this issue because I have installed the plugin et i'm on GLPI 10.0.2
When i want to export the ticket I have the same error.

![image](https://user-images.githubusercontent.com/63521311/200870185-4f1ac5d7-f26a-465e-9b24-1a4df74d5874.png)

**Comment by cedrobert** [comment]
Hello,
I have the same issue with GLPI 9.5.9 and PDF 2.0.1

**Comment by cedrobert** [comment]
thank it's OK for me. In GLPI 9.5 and PDF 2.0.1 with this changes.

**Comment by stdglpi** [comment]
On GLPI 10.0.3 after modified theses two files : 

- inc/itilfollowup.class.php
- inc/ticket.class.php

the error still occurs

**Comment by yllen** [comment]
Mail received from Eusebio

I have replaced my current php files for the new ones you sent me in your last email: "itilfollowup.class.php and ticket.class.php" and then I refresh "Google Chrome Web Explorer"
And I have seen that it work!! PDF conversions are beeing done correctly!
